client/src/main/java/io/pravega/client/connection/impl/ConnectionPoolImpl.java

```diff
@@ -186,6 +186,7 @@ public void pruneUnusedConnections() {
     }
 
     @Override
+    @Synchronized
```

Comment:
Why is this change required?

Reply:
Yeah, there can be a `getClientConnection()` invocation which would update the map after a connection creation while this close() method is invoked.